javaparser-core/src/main/java/com/github/javaparser/utils/TypeUtils.java

```diff
@@ -59,28 +62,36 @@ private static void appendDescriptor(final Class<?> clazz, final StringBuilder s
     }
 
     public static String getPrimitiveTypeDescriptor(final Class<?> clazz) {
-        String descriptor;
-        if (clazz == Void.TYPE) {
-            descriptor = new VoidType().toDescriptor();
-        } else if (clazz == Integer.TYPE) {
-            descriptor = PrimitiveType.intType().toDescriptor();
-        } else if (clazz == Boolean.TYPE) {
-            descriptor = PrimitiveType.booleanType().toDescriptor();
-        } else if (clazz == Byte.TYPE) {
-            descriptor = PrimitiveType.byteType().toDescriptor();
-        } else if (clazz == Character.TYPE) {
-            descriptor = PrimitiveType.charType().toDescriptor();
-        } else if (clazz == Short.TYPE) {
-            descriptor = PrimitiveType.shortType().toDescriptor();
-        } else if (clazz == Double.TYPE) {
-            descriptor = PrimitiveType.doubleType().toDescriptor();
-        } else if (clazz == Float.TYPE) {
-            descriptor = PrimitiveType.floatType().toDescriptor();
-        } else if (clazz == Long.TYPE) {
-            descriptor = PrimitiveType.longType().toDescriptor();
-        } else {
-            throw new AssertionError("Unknown primitive: " + clazz.getName());
+        if (clazz == Void.TYPE || clazz == Void.class) {
+            return new VoidType().toDescriptor();
+        }
+        String className = clazz.getCanonicalName();
```

Comment:
It still seems a bit too complex to me.

If you add 2 methods in Primitive

    public static Optional<Primitive> byBoxedType(String qName) {
            for (Primitive primitive : values()) {
                if (primitive.nameOfBoxedType.equals(qName)) {
                    return Optional.of(primitive);
                }
            }
            return Optional.empty();
        }
        
        public String toDecriptor() {
        	return descriptor;
        }

then you can do it just like that.

    @Test
	void test0() {
		Optional<Primitive> primitive = Primitive.byBoxedType(Integer.class.getSimpleName());
		assertEquals("I", primitive.map(type -> type.toDecriptor()).orElseThrow(() -> new IllegalArgumentException("Unknown primitive type")));
	}

Also don't throw an AssertionException and an optional parameter is bad practice.

Reply:
The newest commit https://github.com/javaparser/javaparser/pull/3819/commits/eaf1f0d06727b9262f3189c01e1f2951e4fc5bc4 basically did what you suggest using the unboxMap and also adding `Primitive.byTypeName` for all the inputs like int.class, boolean.class etc.